src/store/auth.module.js

```diff
@@ -49,19 +49,23 @@ const actions = {
     })
   },
   [CHECK_AUTH] (context) {
-    if (JwtService.getToken()) {
-      ApiService.setHeader()
-      ApiService
-        .get('user')
-        .then(({data}) => {
-          context.commit(SET_AUTH, data.user)
-        })
-        .catch(({response}) => {
-          context.commit(SET_ERROR, response.data.errors)
-        })
-    } else {
-      context.commit(PURGE_AUTH)
-    }
+    return new Promise((resolve, reject) => {
+      if (JwtService.getToken()) {
+        ApiService.setHeader()
+        ApiService
+          .get('user')
+          .then(({data}) => {
+            context.commit(SET_AUTH, data.user)
+            resolve(data)
+          })
+          .catch(({response}) => {
+            context.commit(SET_ERROR, response.data.errors)
+          })
+      } else {
```

Comment:
@igeligel I love that you raised this issue of early returns. I read http://blog.timoxley.com/post/47041269194/avoid-else-return-early to get a fix on the debate. Useful comments on that article. 

Maybe I'm doing it wrong, but my implementation of an early return here would look like:

```
return new Promise((resolve, reject) => {
      const token = JwtService.getToken()
      if (!token) {
        context.commit(PURGE_AUTH)
        return resolve()
      }

      ApiService.setHeader()
      ApiService
        .get('user')
        .then(({data}) => {
          context.commit(SET_AUTH, data.user)
          resolve(data)
        })
        .catch(({response}) => {
          context.commit(SET_ERROR, response.data.errors)
          resolve()
        })
    })
```
Not a fan of that _at all_. 

I think scattering the `return` _adds_ complexity. The benefit of a single column of indentation really doesn't do anything for me here because we don't have nested conditionals anyway. 

If anything I see this supporting the "single exit point" argument, because it would be more difficult for new eyes to debug the scattered return. 

By comparison, we previously had:

```
return new Promise((resolve, reject) => {
      const token = JwtService.getToken()
      if (token) {
        ApiService.setHeader()
        ApiService
          .get('user')
          .then(({data}) => {
            context.commit(SET_AUTH, data.user)
            resolve(data)
          })
          .catch(({response}) => {
            context.commit(SET_ERROR, response.data.errors)
            resolve()
          })
      } else { 
        context.commit(PURGE_AUTH)
        resolve()
      }
    })
```

Would love to read more arguments on this.